Wellthier/View Controllers/WorkoutViewController.m

```diff
@@ -31,10 +42,15 @@ - (void)viewDidLoad {
     self.workoutImageView.file = self.detailWorkout[@"image"];
     PFUser *user = self.detailWorkout[@"author"];
     self.author.text = user[@"displayName"];
+}
+
+- (void) viewWillAppear:(BOOL)animated {
     [self getExercises];
 }
 
 - (void) getExercises {
+    self.arrayOfExercises = [NSArray new];
+    self.filteredExercises = [NSArray new];
```

Comment:
We may not need this two lines

Reply:
these lines are needed, proceeding lines do not work correctly if these are not initialized. i tried

Reply:
Try to use <> to declare the object types when initializing these Data Structure, including NSArray, NSSet and NSDictionary.
Example:
NSArray<Exercise *> *arrayOfExercises = [NSArray new];

Reply:
i cant initialize in the.h file, should move my declarations to the .m file?